Re-factor monolithic pick and place demo

avalenzu opened the pull request:
This (admittedly quite large) PR re-factors the `monolithic_pick_and_place_demo` to allow reuse of its components in a (forthcoming) multi-process version of the demo.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/robotlocomotion/drake/7474)
<!-- Reviewable:end -->

avalenzu commented:
+(status: do not review)  Fixing CI failures

---

Review status: 0 of 25 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-:-Kz-CLUoDbiCM2aO__op:b-k027t3)*
<!-- Sent from Reviewable.io -->

avalenzu commented:
@drake-jenkins-bot linux-xenial-gcc-bazel-experimental please

---

Review status: 0 of 25 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-:-Kz-exH4KRUoQqNfVjE1:b-rm2p6t)*
<!-- Sent from Reviewable.io -->

avalenzu commented:
-(status: do not review) +(status: curate commits before merging) 

---

Review status: 0 of 25 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-:-Kz01uPCYyNHW2fWyLPj:b-63fbpt)*
<!-- Sent from Reviewable.io -->

avalenzu commented:
+@sammy-tri for feature review. The table below shows changes for `dev/` vs non-`dev/` files.

|                                                                                | +   |  -  |
|----------------------------------------------------------------------------------------|----:|----:|
| Non-`dev/` files                                                                               | **304** |**153**  |
| `dev/` files                                                                                     | **1646**  | **796** |

---

Review status: 0 of 25 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-:-Kz020oA3L4B2HXH1SuM:bnhhp7v)*
<!-- Sent from Reviewable.io -->

sammy-tri commented:
Checkpoint.  I think this one might have to be done in stages....  :)

---

Reviewed 17 of 27 files at r1.
Review status: 15 of 25 files reviewed at latest revision, 6 unresolved discussions.

---

*[drake/examples/kuka_iiwa_arm/dev/pick_and_place/lcm_plant.h, line 14 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-Kz9ljpeh9xUyX03jagN:-Kz9ljpeh9xUyX03jagO:b3mkj1t) ([raw file](https://github.com/robotlocomotion/drake/blob/c4cea9b9f52c332aab3cd8e90f82f26ff1fcd06d/drake/examples/kuka_iiwa_arm/dev/pick_and_place/lcm_plant.h#L14)):*
> ```objc
> namespace kuka_iiwa_arm {
> namespace pick_and_place {
> class LcmPlant : public systems::Diagram<double> {
> ```

I know it's in `dev/`, but the name LcmPlant seems sufficiently generic that a class comment would be helpful.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 31 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-Kz9jMtrrwrLB9dbXYQf:-Kz9jMts-BhkyLXlKwRT:b-24l8xe) ([raw file](https://github.com/robotlocomotion/drake/blob/c4cea9b9f52c332aab3cd8e90f82f26ff1fcd06d/drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L31)):*
> ```cpp
>   const double kPitchOffset = 0.0;
> 
>   // Gripper pose should some distance in from the edge of the object, and
> ```

"Gripper pose should some distance in" feels like it's missing a verb or something.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 286 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-Kz9kZcLcNQ90QwE_oMO:-Kz9kZcMw4JbqiS36fWt:b3mf55k) ([raw file](https://github.com/robotlocomotion/drake/blob/c4cea9b9f52c332aab3cd8e90f82f26ff1fcd06d/drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L286)):*
> ```cpp
>       if (!iiwa_move_.ActionStarted()) {
>         X_Wend_effector_0_ = X_Wend_effector_1_;
>         // X_Wend_effector_1_ = ComputeGraspPose(env_state.get_object_pose());
> ```

remove commented out code

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.h, line 51 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-Kz9dFDuRS4biCf5BZIw:-Kz9dFDuRS4biCf5BZIx:b4pjjsu) ([raw file](https://github.com/robotlocomotion/drake/blob/c4cea9b9f52c332aab3cd8e90f82f26ff1fcd06d/drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.h#L51)):*
> ```objc
>   /// Construct a pick and place state machine.  The state machine will move the
>   /// item counter-clockwise around the tables specified in @p configuration.
>   /// If @p loop is true, the state machine will loop through the pick and place
> ```

The parameter `loop` was removed.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/world_state.h, line 46 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-Kz9cW2OIWKFww_VV01C:-Kz9cW2OIWKFww_VV01D:bz7joel) ([raw file](https://github.com/robotlocomotion/drake/blob/c4cea9b9f52c332aab3cd8e90f82f26ff1fcd06d/drake/examples/kuka_iiwa_arm/pick_and_place/world_state.h#L46)):*
> ```objc
>       const Vector3<double>& object_dimensions = Vector3<double>::Zero());
> 
>   // TODO(sam.creasey) We should consider adding an alternate
> ```

I think this TODO might be obsolete now.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/world_state.h, line 64 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-Kz9ca28J5Kxm7rJTOcv:-Kz9ca28J5Kxm7rJTOcw:bpqgb6m) ([raw file](https://github.com/robotlocomotion/drake/blob/c4cea9b9f52c332aab3cd8e90f82f26ff1fcd06d/drake/examples/kuka_iiwa_arm/pick_and_place/world_state.h#L64)):*
> ```objc
>   void HandleObjectStatus(const bot_core::robot_state_t& obj_msg);
> 
>   /// Update the stored object status from @p obj_msg.
> ```

this function doesn't have an `obj_msg` param.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-:-Kz9mFTvQBK3pqKn0Llc:b79hxv6)*
<!-- Sent from Reviewable.io -->

sammy-tri commented:
Reviewed 3 of 27 files at r1.
Review status: 18 of 25 files reviewed at latest revision, 7 unresolved discussions.

---

*[drake/examples/kuka_iiwa_arm/dev/pick_and_place/lcm_planner.h, line 10 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-Kz9sRu2iFd7XLnYPLKz:-Kz9sRu396ufhXYjjLkA:bm7i5fe) ([raw file](https://github.com/robotlocomotion/drake/blob/c4cea9b9f52c332aab3cd8e90f82f26ff1fcd06d/drake/examples/kuka_iiwa_arm/dev/pick_and_place/lcm_planner.h#L10)):*
> ```objc
> namespace examples {
> namespace kuka_iiwa_arm {
> namespace monolithic_pick_and_place {
> ```

should this namespace be `pick_and_place`?  (at any rate it's inconsistent with `lcm_plant.h`)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7474)*
<!-- Sent from Reviewable.io -->

sammy-tri commented:
First pass complete (and more quickly than I expected given the line count)

---

Reviewed 7 of 27 files at r1.
Review status: all files reviewed at latest revision, 14 unresolved discussions.

---

*[drake/examples/kuka_iiwa_arm/dev/monolithic_pick_and_place/BUILD.bazel, line 10 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzA719KmeQhTh9umUr9:-KzA719LeMXeVcx112-J:bra9uiy) ([raw file](https://github.com/robotlocomotion/drake/blob/c4cea9b9f52c332aab3cd8e90f82f26ff1fcd06d/drake/examples/kuka_iiwa_arm/dev/monolithic_pick_and_place/BUILD.bazel#L10)):*
> ```
> )
> load("//tools/lint:lint.bzl", "add_lint_tests")
> load(":pick_and_place_test.bzl", "pick_and_place_test")
> ```

Should this file be deleted if it's no longer used?

---

*[drake/examples/kuka_iiwa_arm/dev/monolithic_pick_and_place/BUILD.bazel, line 34 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzA7Ze1yhsp1kou7XlX:-KzA7Ze2z3jfaXeXjMU-:bdoxqdv) ([raw file](https://github.com/robotlocomotion/drake/blob/c4cea9b9f52c332aab3cd8e90f82f26ff1fcd06d/drake/examples/kuka_iiwa_arm/dev/monolithic_pick_and_place/BUILD.bazel#L34)):*
> ```
> 
> drake_cc_binary(
>     name = "monolithic_pick_and_place_demo",
> ```

Should this gain `add_test_rule = 1` and `test_rule_args = ["--simulation_sec=0.01"]`  (and probably `test_rule_flaky = 1`) like the other  iiwa simulation examples?

---

*[drake/examples/kuka_iiwa_arm/dev/monolithic_pick_and_place/monolithic_pick_and_place_demo.cc, line 26 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzA7snOKqNYVXbtKREO:-KzA7snOKqNYVXbtKREP:b-g6g0kl) ([raw file](https://github.com/robotlocomotion/drake/blob/c4cea9b9f52c332aab3cd8e90f82f26ff1fcd06d/drake/examples/kuka_iiwa_arm/dev/monolithic_pick_and_place/monolithic_pick_and_place_demo.cc#L26)):*
> ```cpp
> #include "drake/systems/lcm/lcm_publisher_system.h"
> #include "drake/systems/primitives/constant_vector_source.h"
> #include "drake/systems/primitives/zero_order_hold.h"
> ```

unused? (aside from the `using` statement)

---

*[drake/examples/kuka_iiwa_arm/dev/monolithic_pick_and_place/test/monolithic_pick_and_place_system_test.cc, line 168 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzA95JVDZLM1L48tbo1:-KzA95JW08ASC-eW1q39:bvg8ml5) ([raw file](https://github.com/robotlocomotion/drake/blob/c4cea9b9f52c332aab3cd8e90f82f26ff1fcd06d/drake/examples/kuka_iiwa_arm/dev/monolithic_pick_and_place/test/monolithic_pick_and_place_system_test.cc#L168)):*
> ```cpp
>           sys->GetSubsystemContext(*plant, simulator.get_context()));
>     }
>     EXPECT_TRUE(plant->is_done(
> ```

BTW it would be fascinating to see the scenario in which this evaluates to false...

---

*[drake/examples/kuka_iiwa_arm/dev/pick_and_place/lcm_plant.cc, line 168 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzA4fortuLnxj9nPPmB:-KzA4fos4ELFt4FEWzXe:b-v6tmxw) ([raw file](https://github.com/robotlocomotion/drake/blob/c4cea9b9f52c332aab3cd8e90f82f26ff1fcd06d/drake/examples/kuka_iiwa_arm/dev/pick_and_place/lcm_plant.cc#L168)):*
> ```cpp
>     const int robot_base_instance_id = iiwa_instances[i].instance_id;
>     const std::string robot_base_name =
>         tree.get_body(tree.FindBaseBodies(robot_base_instance_id).front())
> ```

I feel like calling `get_body` and `get_name` then `FindBody` is a lot more complicated than just calling `get_mutable_body`.  Is there a reason to do it this way?

---

*[drake/examples/kuka_iiwa_arm/dev/pick_and_place/lcm_plant.cc, line 213 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzA57Q6LLEzKbt1jcEJ:-KzA57Q6LLEzKbt1jcEK:b-ifryu3) ([raw file](https://github.com/robotlocomotion/drake/blob/c4cea9b9f52c332aab3cd8e90f82f26ff1fcd06d/drake/examples/kuka_iiwa_arm/dev/pick_and_place/lcm_plant.cc#L213)):*
> ```cpp
>   auto optitrack_encoder =
>       builder.AddSystem<OptitrackEncoder>(frame_name_to_id_map);
>   optitrack_encoder->set_name("optitrack encoder");
> ```

This `set_name` doesn't seem to add much vs. the default generated name

---

*[drake/examples/kuka_iiwa_arm/dev/pick_and_place/lcm_plant.cc, line 243 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzA5N4bjLyYBfknIc54:-KzA5N4bjLyYBfknIc55:bsmd6cj) ([raw file](https://github.com/robotlocomotion/drake/blob/c4cea9b9f52c332aab3cd8e90f82f26ff1fcd06d/drake/examples/kuka_iiwa_arm/dev/pick_and_place/lcm_plant.cc#L243)):*
> ```cpp
>   for (int i = 0; i < kNumIiwas; ++i) {
>     const std::string suffix{"_" + std::to_string(i)};
>     auto iiwa_command_receiver = builder.AddSystem<IiwaCommandReceiver>(7);
> ```

BTW `iiwa_common.h` defines kIiwaArmNumJoints which might be clearer than using the value 7 directly.  (also below on the status sender)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-:-KzA9ao3xlMViybs4dY8:b-tdwfpn)*
<!-- Sent from Reviewable.io -->

avalenzu commented:
Review status: 11 of 27 files reviewed at latest revision, 14 unresolved discussions, some commit checks failed.

---

*[drake/examples/kuka_iiwa_arm/dev/monolithic_pick_and_place/BUILD.bazel, line 10 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzA719KmeQhTh9umUr9:-KzAXw6PDPsL87POmEC0:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/c4cea9b9f52c332aab3cd8e90f82f26ff1fcd06d/drake/examples/kuka_iiwa_arm/dev/monolithic_pick_and_place/BUILD.bazel#L10)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

Should this file be deleted if it's no longer used?
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/dev/monolithic_pick_and_place/BUILD.bazel, line 34 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzA7Ze1yhsp1kou7XlX:-KzAXw_vX8fwgX6DXR-8:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/c4cea9b9f52c332aab3cd8e90f82f26ff1fcd06d/drake/examples/kuka_iiwa_arm/dev/monolithic_pick_and_place/BUILD.bazel#L34)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

Should this gain `add_test_rule = 1` and `test_rule_args = ["--simulation_sec=0.01"]`  (and probably `test_rule_flaky = 1`) like the other  iiwa simulation examples?
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/dev/monolithic_pick_and_place/monolithic_pick_and_place_demo.cc, line 26 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzA7snOKqNYVXbtKREO:-KzKssAkmc-_2YIU0Nal:bn57zyb) ([raw file](https://github.com/robotlocomotion/drake/blob/c4cea9b9f52c332aab3cd8e90f82f26ff1fcd06d/drake/examples/kuka_iiwa_arm/dev/monolithic_pick_and_place/monolithic_pick_and_place_demo.cc#L26)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

unused? (aside from the `using` statement)
</blockquote></details>

Done. Got rid of some other unused headers as well.

---

*[drake/examples/kuka_iiwa_arm/dev/monolithic_pick_and_place/test/monolithic_pick_and_place_system_test.cc, line 168 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzA95JVDZLM1L48tbo1:-KzKszC8kq_xve5k-667:b-ln3dl6) ([raw file](https://github.com/robotlocomotion/drake/blob/c4cea9b9f52c332aab3cd8e90f82f26ff1fcd06d/drake/examples/kuka_iiwa_arm/dev/monolithic_pick_and_place/test/monolithic_pick_and_place_system_test.cc#L168)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

BTW it would be fascinating to see the scenario in which this evaluates to false...
</blockquote></details>

:) This was a hold-over from a previous iteration of this file. Removed.

---

*[drake/examples/kuka_iiwa_arm/dev/pick_and_place/lcm_planner.h, line 10 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-Kz9sRu2iFd7XLnYPLKz:-KzA90fgGkYP1sIlaqOq:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/c4cea9b9f52c332aab3cd8e90f82f26ff1fcd06d/drake/examples/kuka_iiwa_arm/dev/pick_and_place/lcm_planner.h#L10)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

should this namespace be `pick_and_place`?  (at any rate it's inconsistent with `lcm_plant.h`)
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/dev/pick_and_place/lcm_plant.cc, line 168 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzA4fortuLnxj9nPPmB:-KzKt6_Wi3runrCFhNIQ:bl8yfma) ([raw file](https://github.com/robotlocomotion/drake/blob/c4cea9b9f52c332aab3cd8e90f82f26ff1fcd06d/drake/examples/kuka_iiwa_arm/dev/pick_and_place/lcm_plant.cc#L168)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

I feel like calling `get_body` and `get_name` then `FindBody` is a lot more complicated than just calling `get_mutable_body`.  Is there a reason to do it this way?
</blockquote></details>

I didn't realize it at the time, but this was a sneaky way to defeat the `const`-ness of `tree`. I've now replaced this trickery with explicit `const_cast`s.

---

*[drake/examples/kuka_iiwa_arm/dev/pick_and_place/lcm_plant.cc, line 213 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzA57Q6LLEzKbt1jcEJ:-KzKtKpyZkVv_sjDUn0f:bss4a2b) ([raw file](https://github.com/robotlocomotion/drake/blob/c4cea9b9f52c332aab3cd8e90f82f26ff1fcd06d/drake/examples/kuka_iiwa_arm/dev/pick_and_place/lcm_plant.cc#L213)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

This `set_name` doesn't seem to add much vs. the default generated name
</blockquote></details>

Removed.

---

*[drake/examples/kuka_iiwa_arm/dev/pick_and_place/lcm_plant.cc, line 243 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzA5N4bjLyYBfknIc54:-KzKtLr0kBSfZAZNsJep:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/c4cea9b9f52c332aab3cd8e90f82f26ff1fcd06d/drake/examples/kuka_iiwa_arm/dev/pick_and_place/lcm_plant.cc#L243)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

BTW `iiwa_common.h` defines kIiwaArmNumJoints which might be clearer than using the value 7 directly.  (also below on the status sender)
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/dev/pick_and_place/lcm_plant.h, line 14 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-Kz9ljpeh9xUyX03jagN:-KzKtO3jTHeWzf8N5hul:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/c4cea9b9f52c332aab3cd8e90f82f26ff1fcd06d/drake/examples/kuka_iiwa_arm/dev/pick_and_place/lcm_plant.h#L14)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

I know it's in `dev/`, but the name LcmPlant seems sufficiently generic that a class comment would be helpful.
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 31 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-Kz9jMtrrwrLB9dbXYQf:-KzKtQ1bqjtT9duQAGBT:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/c4cea9b9f52c332aab3cd8e90f82f26ff1fcd06d/drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L31)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

"Gripper pose should some distance in" feels like it's missing a verb or something.
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 286 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-Kz9kZcLcNQ90QwE_oMO:-KzKtR-Slt_ZVvX7UKOj:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/c4cea9b9f52c332aab3cd8e90f82f26ff1fcd06d/drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L286)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

remove commented out code
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.h, line 51 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-Kz9dFDuRS4biCf5BZIw:-KzKtRqPN5GwArGs-KVy:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/c4cea9b9f52c332aab3cd8e90f82f26ff1fcd06d/drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.h#L51)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

The parameter `loop` was removed.
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/world_state.h, line 46 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-Kz9cW2OIWKFww_VV01C:-KzKtSVXbdeYr5wIm-3w:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/c4cea9b9f52c332aab3cd8e90f82f26ff1fcd06d/drake/examples/kuka_iiwa_arm/pick_and_place/world_state.h#L46)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

I think this TODO might be obsolete now.
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/world_state.h, line 64 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-Kz9ca28J5Kxm7rJTOcv:-KzKtUQQBEKyxovrlhic:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/c4cea9b9f52c332aab3cd8e90f82f26ff1fcd06d/drake/examples/kuka_iiwa_arm/pick_and_place/world_state.h#L64)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

this function doesn't have an `obj_msg` param.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7474)*
<!-- Sent from Reviewable.io -->

avalenzu commented:
@drake-jenkins-bot linux-xenial-clang-bazel-experimental-everything please

---

Review status: 11 of 27 files reviewed at latest revision, 12 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-:-KzKtYMOHzit8Ih16ML1:b-rs4zsh)*
<!-- Sent from Reviewable.io -->

avalenzu commented:
Addressed @sammy-tri's comments. PTAL (After Thanksgiving :)  )

---

Review status: 11 of 28 files reviewed at latest revision, 12 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-:-KzM5scQdUnMaN6fXr2V:b-1z6bat)*
<!-- Sent from Reviewable.io -->

sammy-tri commented:
LGTM pending platform review of the non-dev bits (not putting the emoji on yet since I don't want reviewable to think platform review is complete)

---

Reviewed 2 of 27 files at r1, 17 of 17 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[drake/examples/kuka_iiwa_arm/dev/monolithic_pick_and_place/test/monolithic_pick_and_place_system_test.cc, line 168 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzA95JVDZLM1L48tbo1:-KzPh7aFKr6VksM7JWTC:b-pxzrmf) ([raw file](https://github.com/robotlocomotion/drake/blob/c4cea9b9f52c332aab3cd8e90f82f26ff1fcd06d/drake/examples/kuka_iiwa_arm/dev/monolithic_pick_and_place/test/monolithic_pick_and_place_system_test.cc#L168)):*
<details><summary><i>Previously, avalenzu (Andres Valenzuela) wrote…</i></summary><blockquote>

:) This was a hold-over from a previous iteration of this file. Removed.
</blockquote></details>

I don't see any changes?

---

*[drake/examples/kuka_iiwa_arm/dev/pick_and_place/lcm_plant.cc, line 168 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzA4fortuLnxj9nPPmB:-KzPgiJrAILilEI0PMTm:b-n559rv) ([raw file](https://github.com/robotlocomotion/drake/blob/c4cea9b9f52c332aab3cd8e90f82f26ff1fcd06d/drake/examples/kuka_iiwa_arm/dev/pick_and_place/lcm_plant.cc#L168)):*
<details><summary><i>Previously, avalenzu (Andres Valenzuela) wrote…</i></summary><blockquote>

I didn't realize it at the time, but this was a sneaky way to defeat the `const`-ness of `tree`. I've now replaced this trickery with explicit `const_cast`s.
</blockquote></details>

I guess one alternative would be to add `get_mutable_tree` to the plant, but this works for me.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-:-KzPhV-4kCgyW9oGp-91:bvnu8sj)*
<!-- Sent from Reviewable.io -->

avalenzu commented:
Review status: 27 of 28 files reviewed at latest revision, all discussions resolved.

---

*[drake/examples/kuka_iiwa_arm/dev/monolithic_pick_and_place/test/monolithic_pick_and_place_system_test.cc, line 168 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzA95JVDZLM1L48tbo1:-KzPy0LLV-u3LyH7M8Mm:b-z4idag) ([raw file](https://github.com/robotlocomotion/drake/blob/c4cea9b9f52c332aab3cd8e90f82f26ff1fcd06d/drake/examples/kuka_iiwa_arm/dev/monolithic_pick_and_place/test/monolithic_pick_and_place_system_test.cc#L168)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

I don't see any changes?
</blockquote></details>

Actually removed now.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7474)*
<!-- Sent from Reviewable.io -->

SeanCurtis-TRI commented:
+@seancurtis-tri  for platform review

You've got a lot of magic numbers that could probably be documented. And a few other small nits/suggestions.  Otherwise <img class="emoji" title=":LGTM:" alt=":LGTM:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> on non-dev stuff.

---

Reviewed 9 of 27 files at r1, 3 of 17 files at r2.
Review status: 27 of 28 files reviewed at latest revision, 14 unresolved discussions.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/action.cc, line 51 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzQCvYmzg2f3Ve57aIV:-KzQCvYmzg2f3Ve57aIW:b-r20w7z) ([raw file](https://github.com/robotlocomotion/drake/blob/33b121849193b4721a2c4d2f1ec50676e055df18/drake/examples/kuka_iiwa_arm/pick_and_place/action.cc#L51)):*
> ```cpp
>   *plan = EncodeKeyFrames(iiwa, time, info, q_mat);
>   StartAction(est_state.get_iiwa_time());
>   duration_ = time.back() + 0.5;
> ```

BTW Magic number could definitely benefit from documentation.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 22 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzQD9wghPbNvsq60q00:-KzQD9wghPbNvsq60q01:bybw3jd) ([raw file](https://github.com/robotlocomotion/drake/blob/33b121849193b4721a2c4d2f1ec50676e055df18/drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L22)):*
> ```cpp
> 
> // Computes the desired end effector pose in the world frame given the object
> // pose in the world frame.
> ```

BTW Given the fact the interface changed, probably worth updating the document.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 25 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzQDGrJl17GMBzLt-N7:-KzQDGrJl17GMBzLt-N8:bcwp2y4) ([raw file](https://github.com/robotlocomotion/drake/blob/33b121849193b4721a2c4d2f1ec50676e055df18/drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L25)):*
> ```cpp
> Isometry3<double> ComputeGraspPose(const Isometry3<double>& X_WObj,
>                                    Vector3<double> object_dimensions) {
>   // Sets desired end effector location to be 19cm behind the object,
> ```

FYI You might consider re-wording this so that you refer to the value below, but not copy it. That way, if the value changes, the documentation doesn't have to change as well.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 34 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzQDqem9iwIyf5E89Nn:-KzQDqem9iwIyf5E89No:b-xuip73) ([raw file](https://github.com/robotlocomotion/drake/blob/33b121849193b4721a2c4d2f1ec50676e055df18/drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L34)):*
> ```cpp
>   // and should be rotated about the y-axis by pitch_offset.
>   Isometry3<double> X_OG{Isometry3<double>::Identity()};
>   X_OG.rotate(AngleAxis<double>(kPitchOffset, Vector3<double>::UnitY()));
> ```

BTW Generally, I'd like to see the semantics of the magic numbers explained (even if not the specific values).

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 40 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzQE6AUeoom4Lu5EUwB:-KzQE6AUeoom4Lu5EUwC:b-7mjmu7) ([raw file](https://github.com/robotlocomotion/drake/blob/33b121849193b4721a2c4d2f1ec50676e055df18/drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L40)):*
> ```cpp
>           -0.5 * object_dimensions.x() + 0.07 * std::cos(kPitchOffset), 0));
> 
>   // The gripper is rotated relative to the end effector link.
> ```

FYI In the previous "paragraph", you describe the grasp frame G relative to the object frame (giving rise go `X_OG`). Here, you reverse it talking about the gripper relative to the end effector, but still get `X_GE`.  Maybe turn the rhetoric around?

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 42 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzQDn3PRhJYH_nT48Fo:-KzQDn3PRhJYH_nT48Fp:b-yobtwn) ([raw file](https://github.com/robotlocomotion/drake/blob/33b121849193b4721a2c4d2f1ec50676e055df18/drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L42)):*
> ```cpp
>   // The gripper is rotated relative to the end effector link.
>   Isometry3<double> X_GE{Isometry3<double>::Identity()};
>   X_GE.rotate(Eigen::AngleAxisd(0.39269908, Eigen::Vector3d::UnitX()));
> ```

FYI A *very* ugly magic number.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 112 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzQEZ8qwblyajg364nl:-KzQEZ8qwblyajg364nm:bbxfcu6) ([raw file](https://github.com/robotlocomotion/drake/blob/33b121849193b4721a2c4d2f1ec50676e055df18/drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L112)):*
> ```cpp
> bool ComputeInitialAndFinalObjectPoses(
>     const WorldState& env_state,
>     std::pair<Isometry3<double>, Isometry3<double>>* X_WO_initial_and_final) {
> ```

FYI Why a pair? If you were simply returning the pair, that'd make sense. But as long as you're passing it in this way, why not break each pose individually, by name?

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 114 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzQFebwUuYEVn7Ci_91:-KzQFebwUuYEVn7Ci_92:b-pl69r5) ([raw file](https://github.com/robotlocomotion/drake/blob/33b121849193b4721a2c4d2f1ec50676e055df18/drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L114)):*
> ```cpp
>     std::pair<Isometry3<double>, Isometry3<double>>* X_WO_initial_and_final) {
>   // W -- planning World frame, coincides with kuka base frame.
>   // S -- Sensor world frame
> ```

FYI You don't get two "world" frames.  Probably just the "fixed sensor frame".

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 117 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzQEkmDAM_Dv5-p1_fq:-KzQEkmDAM_Dv5-p1_fr:b-p290r3) ([raw file](https://github.com/robotlocomotion/drake/blob/33b121849193b4721a2c4d2f1ec50676e055df18/drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L117)):*
> ```cpp
>   // O -- Object frame
>   // T -- Table frame
>   Isometry3<double> X_WS{env_state.get_iiwa_base().inverse()};
> ```

FYI This is the most counter-intuitive line of code. How is the iiwa base's inverse `X_WS`? Particularly since `W` was described as to "coincide with the kuka base frame" which *sounds* like `get_iiwa_base()`. The sensor, in this case, is presumably a *fixed* frame (e.g., optitrack)? 

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 233 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzQH7CgaiAa0-I3OpxY:-KzQH7CgaiAa0-I3OpxZ:b-1ns37a) ([raw file](https://github.com/robotlocomotion/drake/blob/33b121849193b4721a2c4d2f1ec50676e055df18/drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L233)):*
> ```cpp
>           const Isometry3<double>& pose = env_state.get_object_pose();
>           drake::log()->info("Object at: {} {}", pose.translation().transpose(),
>                              math::rotmat2rpy(pose.rotation()).transpose());
> ```

FYI A smidgen of documentation on why the rotation is inverted would be nice.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 250 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzQHGqk2sIxs8zc055j:-KzQHGqk2sIxs8zc055k:b-z6zxd0) ([raw file](https://github.com/robotlocomotion/drake/blob/33b121849193b4721a2c4d2f1ec50676e055df18/drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L250)):*
> ```cpp
>         // Computes the desired end effector pose in the world frame to be
>         // kPreGraspHeightOffset above the object.
>         const Isometry3<double>& X_WO = X_WO_initial_and_final_.first;
> ```

FYI Regarding my earlier comment. Given that the first thing you do with the pair is extract one of the values into something readable, that seems to be one more reason *not* to write to a pair.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/world_state.h, line 48 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzQICQW_V99KJqHzl3K:-KzQICQW_V99KJqHzl3L:b-x46tvb) ([raw file](https://github.com/robotlocomotion/drake/blob/33b121849193b4721a2c4d2f1ec50676e055df18/drake/examples/kuka_iiwa_arm/pick_and_place/world_state.h#L48)):*
> ```objc
>   ~WorldState();
> 
>   /// Update the stored iiwa status from @p iiwa_msg.
> ```

BTW New parameter suggests a modified doxygen comment.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/test/pick_and_place_state_machine_test.cc, line 46 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzQISNmXN39ZuXfJakt:-KzQISNmXN39ZuXfJaku:bea6rsy) ([raw file](https://github.com/robotlocomotion/drake/blob/33b121849193b4721a2c4d2f1ec50676e055df18/drake/examples/kuka_iiwa_arm/pick_and_place/test/pick_and_place_state_machine_test.cc#L46)):*
> ```cpp
>   WorldState world_state(FindResourceOrThrow(kIiwaUrdf), "iiwa_link_ee",
>                          planner_configuration.num_tables,
>                          planner_configuration.target_dimensions);
> ```

FYI might be worth having an interface that simply *consumes* the planner configuration and extracts multiple fields itself.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/test/pick_and_place_state_machine_test.cc, line 75 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzQIaKeMKeLetQM_wUw:-KzQIaKeMKeLetQM_wUx:b-cbp61l) ([raw file](https://github.com/robotlocomotion/drake/blob/33b121849193b4721a2c4d2f1ec50676e055df18/drake/examples/kuka_iiwa_arm/pick_and_place/test/pick_and_place_state_machine_test.cc#L75)):*
> ```cpp
> 
>   Isometry3<double> X_WT;
>   X_WT.translation() = Eigen::Vector3d(0.80, 0.36, 0);
> ```

FYI Fun magic numbers....

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-:-KzQImvX2jFFFo4wONUN:b-koq48b)*
<!-- Sent from Reviewable.io -->

sammy-tri commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 14 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-:-KzZIQthLydiT2flfqGx:bnfp4nl)*
<!-- Sent from Reviewable.io -->

avalenzu commented:
Addressed reviewer comments and squashed. -(status: curate commits before merging) 

---

Review status: 23 of 28 files reviewed at latest revision, 14 unresolved discussions.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/action.cc, line 51 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzQCvYmzg2f3Ve57aIV:-KzRWeQq3-vRQLYlwcWm:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/33b121849193b4721a2c4d2f1ec50676e055df18/drake/examples/kuka_iiwa_arm/pick_and_place/action.cc#L51)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

BTW Magic number could definitely benefit from documentation.
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 22 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzQD9wghPbNvsq60q00:-KzRXLCz80FdhA0SfBLx:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/33b121849193b4721a2c4d2f1ec50676e055df18/drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L22)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

BTW Given the fact the interface changed, probably worth updating the document.
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 25 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzQDGrJl17GMBzLt-N7:-KzRahqiWWW34Z04HOUL:b-2xit33) ([raw file](https://github.com/robotlocomotion/drake/blob/33b121849193b4721a2c4d2f1ec50676e055df18/drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L25)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

FYI You might consider re-wording this so that you refer to the value below, but not copy it. That way, if the value changes, the documentation doesn't have to change as well.
</blockquote></details>

Rendered moot.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 34 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzQDqem9iwIyf5E89Nn:-Kzoi9G_PEikIw_iVp1s:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/33b121849193b4721a2c4d2f1ec50676e055df18/drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L34)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

BTW Generally, I'd like to see the semantics of the magic numbers explained (even if not the specific values).
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 40 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzQE6AUeoom4Lu5EUwB:-Kzok4XM4kQmrkQLIh9T:b-pgd13d) ([raw file](https://github.com/robotlocomotion/drake/blob/33b121849193b4721a2c4d2f1ec50676e055df18/drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L40)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

FYI In the previous "paragraph", you describe the grasp frame G relative to the object frame (giving rise go `X_OG`). Here, you reverse it talking about the gripper relative to the end effector, but still get `X_GE`.  Maybe turn the rhetoric around?
</blockquote></details>

Done. Also added a `TODO` for moving this information into a gripper model.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 42 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzQDn3PRhJYH_nT48Fo:-KzRc3e9BDIJkYfk4htc:b-mrat00) ([raw file](https://github.com/robotlocomotion/drake/blob/33b121849193b4721a2c4d2f1ec50676e055df18/drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L42)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

FYI A *very* ugly magic number.
</blockquote></details>

Do you have something against π/8? ;)  Hopefully less ugly now.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 112 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzQEZ8qwblyajg364nl:-KzlOLZ7fzbfjxxWtaK7:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/33b121849193b4721a2c4d2f1ec50676e055df18/drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L112)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

FYI Why a pair? If you were simply returning the pair, that'd make sense. But as long as you're passing it in this way, why not break each pose individually, by name?
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 114 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzQFebwUuYEVn7Ci_91:-KzlOxDVZbeHquKpGRiq:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/33b121849193b4721a2c4d2f1ec50676e055df18/drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L114)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

FYI You don't get two "world" frames.  Probably just the "fixed sensor frame".
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 117 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzQEkmDAM_Dv5-p1_fq:-KzlPpoDnPYjpj_Cis0V:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/33b121849193b4721a2c4d2f1ec50676e055df18/drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L117)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

FYI This is the most counter-intuitive line of code. How is the iiwa base's inverse `X_WS`? Particularly since `W` was described as to "coincide with the kuka base frame" which *sounds* like `get_iiwa_base()`. The sensor, in this case, is presumably a *fixed* frame (e.g., optitrack)? 
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 233 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzQH7CgaiAa0-I3OpxY:-KzobOrLAhkUusIwuJkX:b8j6ftq) ([raw file](https://github.com/robotlocomotion/drake/blob/33b121849193b4721a2c4d2f1ec50676e055df18/drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L233)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

FYI A smidgen of documentation on why the rotation is inverted would be nice.
</blockquote></details>

The rotation isn't inverted, the roll-pitch-yaw vector is transposed so that it displays all on one row.jj

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 250 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzQHGqk2sIxs8zc055j:-KzobZv6PD1iAC-jh7eW:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/33b121849193b4721a2c4d2f1ec50676e055df18/drake/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L250)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

FYI Regarding my earlier comment. Given that the first thing you do with the pair is extract one of the values into something readable, that seems to be one more reason *not* to write to a pair.
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/world_state.h, line 48 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzQICQW_V99KJqHzl3K:-KzocESI-5ifGlROmTwE:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/33b121849193b4721a2c4d2f1ec50676e055df18/drake/examples/kuka_iiwa_arm/pick_and_place/world_state.h#L48)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

BTW New parameter suggests a modified doxygen comment.
</blockquote></details>

Done.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/test/pick_and_place_state_machine_test.cc, line 46 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzQISNmXN39ZuXfJakt:-Kzoe2OK-OFkM8ULe0qc:b-4cuziu) ([raw file](https://github.com/robotlocomotion/drake/blob/33b121849193b4721a2c4d2f1ec50676e055df18/drake/examples/kuka_iiwa_arm/pick_and_place/test/pick_and_place_state_machine_test.cc#L46)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

FYI might be worth having an interface that simply *consumes* the planner configuration and extracts multiple fields itself.
</blockquote></details>

Deferred.

---

*[drake/examples/kuka_iiwa_arm/pick_and_place/test/pick_and_place_state_machine_test.cc, line 75 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-KzQIaKeMKeLetQM_wUw:-KzohV3F17C2GeKSFZuZ:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/33b121849193b4721a2c4d2f1ec50676e055df18/drake/examples/kuka_iiwa_arm/pick_and_place/test/pick_and_place_state_machine_test.cc#L75)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

FYI Fun magic numbers....
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-:-Kzp3vnEL6UcZPnMq0Pd:b-7zpjg2)*
<!-- Sent from Reviewable.io -->

avalenzu commented:
@drake-jenkins-bot linux-xenial-clang-bazel-experimental-memcheck-lsan please

---

Review status: 23 of 28 files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-:-KzqT301lrFmUE0cqB-j:bwqgz7o)*
<!-- Sent from Reviewable.io -->

avalenzu commented:
All comments addressed. CI is happy. @seancurtis-tri, for some reason Reviewable didn't pick up on the "FYI" portion of your comments. I tried dismissing you from one of those discussions (sorry) but then Reviewable said the discussion needed another participant. Could I ask you to change your disposition on the remaining discussions? Thanks!

---

Review status: 23 of 28 files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-:-KzsJebE8NcXllHpDbP-:bt8tcb3)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
> I tried dismissing you from one of those discussions (sorry) but then Reviewable said the discussion needed another participant.

The way that you can self-resolve this is to click on the face circle and then explicitly mark _yourself_ as satisfied.

avalenzu commented:
Oh, I feel foolish now! Thanks @jwnimmer-tri !

---

Review status: 23 of 28 files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7474#-:-KzstP72tj6bCRt2HDHs:b-u38u93)*
<!-- Sent from Reviewable.io -->
